src/contrib/cluster/Akka.Cluster.Sharding/ShardCoordinator.cs

```diff
@@ -1188,7 +1191,26 @@ public CoordinatorState Updated(IDomainEvent e)
                             if (!Regions.TryGetValue(message.Region, out var shardRegions))
                                 throw new ArgumentException($"Region {message.Region} not registered: {this}", nameof(e));
                             if (Shards.ContainsKey(message.Shard))
-                                throw new ArgumentException($"Shard {message.Shard} already allocated: {this}", nameof(e));
+                            {
+                                if(!isRecovering)
+                                    throw new ArgumentException($"Shard {message.Shard} already allocated: {this}",
+                                        nameof(e));
+                                
+                                // per https://github.com/akkadotnet/akka.net/issues/5604
+                                // we're going to allow new value to overwrite previous
+                                var newRegions = Regions;
```

Comment:
A couple questions: 1) Is this safe all the times?; 2) Will there be a way to disable this new behavior?

Reply:
1. Not sure - the alternative is defaulting the current behavior, which is that the `ShardCoordinator` blows up and the cluster needs to be restarted and all of the `ShardCoordinator` data deleted. Given that, I think this probably is safer than the current defaults. The worst case scenario I can imagine here is that the `Shard` was _actually_ allocated to two different nodes, but that would if that were the case then the sharding system would already be in very bad shape (i.e. violating its consistency rules) and this would allow the newest home for the shard to supersede the old one, which would stop receiving message traffic. In the logs where I saw this occuring it was duplicate records for the same `Shard` in the same `ShardRegion IActorRef` - so this change would just make the recovery idempotent.

2. We could add it, but I don't think it should be necessary.